searchlib/src/vespa/searchlib/features/dotproductfeature.cpp

```diff
@@ -332,16 +333,15 @@ template <typename A>
 DotProductByCopyExecutor<A>::~DotProductByCopyExecutor() = default;
 
 template <typename A>
-size_t
-DotProductByCopyExecutor<A>::getAttributeValues(uint32_t docId, const AT * & values)
+vespalib::ConstArrayRef<typename DotProductByCopyExecutor<A>::AT>
+DotProductByCopyExecutor<A>::getAttributeValues(uint32_t docId)
 {
     size_t count = this->_attribute->getAll(docId, &_copy[0], _copy.size());
     if (count > _copy.size()) {
         _copy.resize(count);
         count = this->_attribute->getAll(docId, &_copy[0], _copy.size());
     }
-    values = reinterpret_cast<const AT *>(&_copy[0]);
-    return count;
+    return vespalib::ConstArrayRef(reinterpret_cast<const AT *>(&_copy[0]), count);
```

Comment:
Consider using std::vector::data() to access the raw pointer.

Reply:
Done.